rfcs/text/0030/file_description.yml

```diff
@@ -0,0 +1,271 @@
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+- name: fd
+  title: File description
+  group: 2
+  short: These fields contain information about a file descriptor.
+  description: >
+    These fields contain information about a file descriptor.
+
+    In Unix and Unix-like computer operating systems, a file descriptor (FD, less frequently fildes) is a unique identifier (handle) for a file or other input/output resource, such as a pipe or network socket.
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - at: process
+        as: fds
+        short_override: An array of file descriptors for the process.
+
+  fields:
+    - name: type
+      level: extended
+      type: keyword
+      short: The type of file descriptor.
+      description: >
+        The type of file descriptor.
+
+        One of: file, char_device, pipe, fifo, socket, epoll, signal, inotify, bpf, event, perf_event, timer, userfault, namespace, io_uring
+      example: char_device
+
+    - name: descriptor
+      level: extended
+      type: short
+      short: The integer representation of the FD.
+      description: >
+        File descriptors typically have non-negative integer values, with negative values being reserved to indicate "no value" or error conditions.
+
+        File descriptors are a part of the POSIX API. Each Unix process (except perhaps daemons) should have three standard POSIX file descriptors, corresponding to the three standard streams: 0 - Standard input (stdin), 1 - Standard output (stdout), 2 - Standard error (stderr).
+
+      example: 1
+
+    - name: char_device
+      level: extended
+      type: object
+      short: Character device major and minor numbers.
+      description: >
+        A Character ('c') Device is one with which the Driver communicates by sending and receiving single characters (bytes, octets).
+
+        Examples for Character Devices: serial ports, parallel ports, sounds cards.
+
+        This field is only populated if fd.type: char_device
+
+    - name: char_device.major
+      level: extended
+      type: long
+      short: The major character device number.
+      description: >
+        The major number identifies the driver associated with the device.
+      example: 1
+
+    - name: char_device.minor
+      level: extended
+      type: long
+      short: The minor character device number.
+      description: >
+        The minor number is used only by the driver specified by the major number; other parts of the kernel don’t use it, and merely pass it along to the driver. It is common for a driver to control several devices; the minor number provides a way for the driver to differentiate among them.
+      example: 128
+
+    - name: file
+      level: extended
+      type: object
```

Comment:
It might make sense to re-use File type fieldset here. Though we would need to add 'block_device' as a new property of file.

Reply:
I would definitely lean that direction as I think it would "simplify" the schema